flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/RestoreTestBase.java

```diff
@@ -81,9 +81,22 @@
 public abstract class RestoreTestBase implements TableTestProgramRunner {
 
     private final Class<? extends ExecNode> execNodeUnderTest;
+    private final AfterRestoreState afterRestoreState;
 
     protected RestoreTestBase(Class<? extends ExecNode> execNodeUnderTest) {
         this.execNodeUnderTest = execNodeUnderTest;
+        this.afterRestoreState = AfterRestoreState.FINITE_AFTER_RESTORE;
+    }
+
+    protected RestoreTestBase(
+            Class<? extends ExecNode> execNodeUnderTest, AfterRestoreState state) {
+        this.execNodeUnderTest = execNodeUnderTest;
+        this.afterRestoreState = state;
+    }
+
+    protected enum AfterRestoreState {
```

Comment:
nit: `AfterRestoreState` -> `AfterRestoreSource`

The enum name already contains the `AfterRestore` prefix, we can drop it from the enum values:

```
AfterRestoreSource.FINITE or AfterRestoreSource.INFINITE
```

Reply:
https://github.com/apache/flink/pull/23728/files#diff-a815488ad1ad3ed866421e69b53f1c0c07cbcf16dac7e6fe357f3bb87d2e20baR101